docs/tab-widgets/distributed-trace-send.asciidoc

```diff
@@ -0,0 +1,162 @@
+// ***************************************************
+
+// tag::net[]
+
+1. Serialize the distributed tracing context of the active transaction or span with
+{apm-dotnet-ref}/public-api.html#api-current-transaction[`CurrentTransaction`] or
+{apm-dotnet-ref}/public-api.html#api-current-span[`CurrentSpan`].
+
+2. Send the serialized context the receiving service.
+
+Example:
+
+[source,csharp]
+----
+string outgoingDistributedTracingData =
+    (Agent.Tracer.CurrentSpan?.OutgoingDistributedTracingData
+        ?? Agent.Tracer.CurrentTransaction?.OutgoingDistributedTracingData)?.SerializeToString();
+// Now send `outgoingDistributedTracingData` to the receiving service
+----
+
+// end::net[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::node[]
+
+1. Start a transaction with {apm-node-ref}/agent-api.html#apm-start-transaction[`apm.startTransaction()`],
+or a span with {apm-node-ref}/agent-api.html#apm-start-span[`apm.startSpan()`].
+
+2. Get the serialized `traceparent` string of the started transaction/span with
+{apm-node-ref}/agent-api.html#apm-current-traceparent[`currentTraceparent`].
+
+3. Encode the `traceparent` and send it to the receiving service inside your regular request.
+
+Example using raw UDP to communicate between two services, A and B:
+
+[source,js]
+----
+agent.startTransaction('my-service-a-transaction'); <1>
+const traceparent = agent.currentTraceparent; <2>
+sendMetadata(`traceparent: ${traceparent}\n`); <3>
+----
+<1> Start a transaction
+<2> Get the current `traceparent`
+<3> Send the `traceparent` as a header to service B.
+
+// end::node[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::python[]
+
+1. Start a transaction with {apm-py-ref}/api.html#client-api-begin-transaction[`begin_transaction()`].
+
+2. Get the `trace_parent` of the active transaction.
+
+3. Send the `trace_parent` to the receiving service.
+
+Example:
+
+[source,python]
+----
+client.begin_transaction('new-transaction')<1>
+
+elasticapm.get_trace_parent_header('new-transaction') <2>
+
+# Send `trace_parent_str` to another service
+----
+<1> Start a new transaction
+<2> Return the string representation of the current transaction's TraceParent object
+// end::python[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::ruby[]
+
+// halp
+
+Start a transaction or span with
+{apm-ruby-ref}/api.html#api-agent-start_transaction[`start_transaction`] or
+{apm-ruby-ref}/api.html#api-agent-start_span[`start_span`].
+
+Get the active `trace_context`.
+
+Send the `trace_context` to the receiving service.
+
+[source,ruby]
+----
+ElasticAPM.start_transaction('Name') <1>
+# Get the trace_context and save to variable
+# Send the trace_context to another service
+----
+<1> Start a transaction
+
+// end::ruby[]
```

Comment:
@elastic/apm-agent-ruby (part two)

These docs are for manually propagating the traceparent in a **receiving** service

Reply:
1. Start a span
1. Get its TraceContext and use it to generate a Traceparent header

```ruby
ElasticAPM.with_span "Name" do |span|
  header = span.trace_context.traceparent.to_header
  # etc ...
end
```

(This is a _sending_ service, see above)